services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/AutoRerouteSpec.groovy

```diff
@@ -237,6 +313,54 @@ class AutoRerouteSpec extends BaseSpecification {
         }
     }
 
+    def "Flow in 'Down' status tries to reroute when connecting a new switch"() {
+        requireProfiles("virtual")
+
+        given: "With-intermediate-switch flow with one alternate path at least"
+        def flow = withIntermediateSwitchFlow(true)
+        def allPaths = db.getPaths(flow.source.datapath, flow.destination.datapath)*.path.sort { it.size() }
```

Comment:
This is a heavy operation which is performed twice (first time inside withIntermediateSwitchFlow)

Reply:
Oops, will fix.